mindmeld/text_preparation/text_preparation_pipeline.py

```diff
@@ -0,0 +1,530 @@
+        return [t["text"] for t in self.tokenize(normalized_text)]
+
+    def tokenize(self, text):
+        """
+        Args:
+            text (str): Input text.
+        Returns:
+            tokens (List[str]): List of tokens.
+        """
+        return TextPreparationPipeline.tokenize_around_mindmeld_annotations(
+            text=text, function=self.tokenizer.tokenize
+        )
+
+    def stem_word(self, word):
+        """
+        Gets the stem of a word. For example, the stem of the word 'fishing' is 'fish'.
+
+        Args:
+            words (List[str]): List of words to stem.
+
+        Returns:
+            stemmed_words (List[str]): List of stemmed words.
+        """
+        return self.stemmer.stem_word(word)
+
+    @staticmethod
+    def find_mindmeld_annotation_re_matches(text):
+        """
+        Args:
+            text (str): The string to find mindmeld annotation instances
+                (" {entity_text|entity_type} ")
+        Returns:
+            matches (List[sre.SRE_Match object]): Regex match objects.
+        """
+        return list(MINDMELD_ANNOTATION_PATTERN.finditer(text))
+
+    @staticmethod
+    def modify_around_annotations(text, function):
+        """Applied a function around the mindmeld annotation.
+
+        function(pre_entity_text) + { + function(entity_text) + |entity_name}
+            + function(post_entity_text)
+
+        Args:
+            text (str): Original sentence with markup to modify.
+            function (function): Function to apply around the annotation
+        Returns:
+            modified_text (str): Text modified around annotations.
+        """
+        matches = TextPreparationPipeline.find_mindmeld_annotation_re_matches(text)
+
+        modified_text = []
+        prev_entity_end = 0
+
+        for match in matches:
+            entity_start, entity_end = match.span()
+
+            # Adds "function(pre_entity_text)" {..
+            text_before_entity = text[prev_entity_end:entity_start]
+            modified_text.append(function(text_before_entity))
+
+            # Modify the Inner Entity Text
+            entity_text_start, entity_text_end = match.span(1)
+            entity_text = match.group(1)
+
+            # Adds " {"
+            modified_text.append(text[entity_start:entity_text_start])
+
+            # Adds function(entity_text)
+            modified_text.append(function(entity_text))
+
+            # Adds "|"
+            modified_text.append(text[entity_text_end:entity_end])
+
+            # Update the previous entity ending index
+            prev_entity_end = entity_end
+
+        if prev_entity_end < len(text):
+            # Adds the remainder of the text after the last end brace } "function(post_entity_text)"
+            modified_text.append(function(text[prev_entity_end : len(text)]))
+
+        return "".join(modified_text)
+
+    @staticmethod
+    def tokenize_around_mindmeld_annotations(text, function):
```

Comment:
Why do we have to configure the tokenize function? Seems like we can just use self.tokenizer, no?

Reply:
Yeah, good point, I'll make that change